circe-checksum/build.gradle

```diff
@@ -0,0 +1,45 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+plugins {
+    id 'java'
+}
+
+dependencies {
+    implementation depLibs.guava
+    implementation depLibs.nettyBuffer
+    implementation depLibs.slf4j
+    testImplementation depLibs.junit
+    testImplementation depLibs.mockito
+}
+
+task generateJniHeaders(type:Exec) {
+    ext {
+        javahOutputDir = "$buildDir/javahGenerated"
+    }
+    dependsOn classes
+    def classpath = sourceSets.main.output.classesDirs.join(":")
```

Comment:
This is problematic since it is done during Gradle's configuration phase.

Reply:
I am not sure I understood this, as what you mean by gradle's configuration phase, and why it is problematic.

Reply:
It mainly impacts the performance of the build, explained in https://docs.gradle.org/current/userguide/performance.html#configuration . However it's possible that in case there isn't a real performance implication when accessing `sourceSets.main.out.classesDirs`. If it triggers dependency resolution, it would be an actual problem.

Another performance related reason for specifying the Gradle tasks in a certain way is to make the build compatible with Gradle Build Cache. That aspect is covered in https://docs.gradle.org/current/userguide/build_cache.html#sec:task_output_caching_inputs . Using Gradle Build Cache doesn't require Gradle Enterprise. For GitHub Actions CI builds, It's possible to use the Gradle remote build cache with the GitHub Actions Cache as the cache backend with  https://github.com/cirruslabs/http-cache-action .

Reply:
The way to investigate possible configuration time issues is to do a build scan for the "help" task on 2. execution:
```
./gradlew help
./gradlew help --scan
```
I created a build scan for this PR (rev a5f28e1) here: https://gradle.com/s/m7ae4q3ulill4 .

The performance->configuration shows the break down for configuration.The https://scans.gradle.com/s/m7ae4q3ulill4/performance/dependency-resolution shows that there isn't any dependency resolution happening during configuration time, so from that perspective it's fine to access `sourceSets.main.out.classesDirs` during configuration time.

The remaining issue is the possible cacheability and correctness of the build. In Gradle, it's considered a bug in the build or one of the plugins if you have to use "clean" to get correct results. The task inputs and outputs should be properly declared primarily for the correctness & reproduceability of the build, but also to be able to get more advantages such as cacheability.